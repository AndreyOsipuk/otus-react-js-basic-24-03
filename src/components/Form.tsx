import { FC, useState } from 'react'

type Props = {
  message: string
}

export const Form: FC<Props> = (props) => {
  const [value, setValue] = useState(props.message)

  const handleChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
    setValue(ev.target.value)
  }

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
  }

  return <form onSubmit={handleSubmit}>
    <input type="text" onChange={handleChange} value={value} />
    <button>Submit</button>
  </form>
}