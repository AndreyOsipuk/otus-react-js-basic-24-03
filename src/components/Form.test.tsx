import React from 'react';
import { render, screen } from '@testing-library/react';
import { Form } from './Form';

it('renders learn react link', () => {
  render(<Form message='test' />);

  const element = screen.getByTestId<HTMLInputElement>('input')

  element.value = 'test'
  expect(element.value).toBe('test');
});

it('render with snapshot', () => {
  const { asFragment } = render(<Form message="test" />);
  expect(asFragment()).toMatchSnapshot();
});

  // it('render component with text', () => {
  //   render(<Button>test</Button>);

  //   expect(screen.getByText(/test/)).toBeInTheDocument();
  // });

  // it('render multiply components', () => {
  //   render(
  //     <>
  //       <Button>test</Button>
  //       <Button>test</Button>
  //     </>
  //   );

  //   expect(screen.queryAllByRole('button').length).toBe(2);
  // });

  // it('button is disabled', () => {
  //   render(<Button disabled>test</Button>);

  //   expect(screen.getByText(/test/)).toBeDisabled();
  // });

  // // it('button have style color red', () => {
  // //   render(<Button label="test" />);

  // //   expect(screen.getByText(/test/)).toHaveStyle({ color: 'red' });
  // // });

  // it('button click with userEvent', async () => {
  //   const mockHandler = jest.fn();
  //   render(<Button click={mockHandler}>test</Button>);

  //   await userEvent.click(screen.getByText(/test/));

  //   expect(mockHandler).toHaveBeenCalledTimes(1);
  // });

  // it('button async click', async () => {
  //   const mockHandler = jest.fn();
  //   render(<Button click={() => setTimeout(mockHandler, 1500)}>test</Button>);

  //   await userEvent.click(screen.getByText(/test/));

  //   await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
  //     timeout: 1600,
  //   });

  //   // await screen.findByText('Clicked twice', { timeout: 1000 });
  // });

  // it('test example', async () => {
  //   const onChange = jest.fn();
  //   render(<input type="checkbox" onChange={onChange} />);

  //   const checkbox = screen.getByRole('checkbox');

  //   await userEvent.dblClick(checkbox);
  //   expect(onChange).toHaveBeenCalledTimes(2);
  //   expect(checkbox).not.toBeChecked();
  // });
