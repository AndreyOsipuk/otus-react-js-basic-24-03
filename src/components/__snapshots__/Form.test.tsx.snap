// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="input"
      type="text"
      value="test"
    />
    <button>
      Submitt
    </button>
  </form>
</DocumentFragment>
`;
